ydb/src/credentials.rs

```diff
@@ -147,6 +151,163 @@ impl Credentials for CommandLineYcToken {
     }
 }
 
+/// Get service account credentials instance
+/// service account key should be:
+/// - in the local file and YDB_SERVICE_ACCOUNT_KEY_FILE_CREDENTIALS environment variable should point to it
+/// - in the local file and it's path is specified
+/// - in the json format string
+///
+/// Example:
+/// ```
+/// use ydb::ServiceAccountCredentials;
+/// let cred = ServiceAccountCredentials::from_env();
+/// ```
+/// or
+/// ```
+/// use ydb::ServiceAccountCredentials;
+/// let json = "....";
+/// let cred = ServiceAccountCredentials::from_json(json);
+/// ```
+/// or
+/// ```
+/// use ydb::ServiceAccountCredentials;
+/// let cred = ServiceAccountCredentials::new("service_account_id", "key_id", "private_key");
+/// ```
+/// or
+/// ```
+/// use ydb::ServiceAccountCredentials;
+/// let cred = ServiceAccountCredentials::from_file("/path/to/file");
+/// ```
+pub struct ServiceAccountCredentials {
+    private_key: SecretString,
+    service_account_id: String,
+    key_id: String,
+}
+
+impl ServiceAccountCredentials {
+    pub fn new(
+        service_account_id: impl Into<String>,
+        key_id: impl Into<String>,
+        private_key: impl Into<String>,
+    ) -> Self {
+        Self {
+            private_key: SecretString::new(private_key.into()),
+            service_account_id: service_account_id.into(),
+            key_id: key_id.into(),
+        }
+    }
+
+    pub fn from_env() -> YdbResult<Self> {
+        let path = std::env::var("YDB_SERVICE_ACCOUNT_KEY_FILE_CREDENTIALS")?;
+
+        ServiceAccountCredentials::from_file(path)
+    }
+
+    pub fn from_file(path: impl AsRef<std::path::Path>) -> YdbResult<Self> {
+        let json_key = std::fs::read_to_string(path)?;
+        ServiceAccountCredentials::from_json(&json_key)
+    }
+
+    pub fn from_json(json_key: &str) -> YdbResult<Self> {
+        #[derive(Debug, Serialize, Deserialize)]
+        struct JsonKey {
+            public_key: String,
+            private_key: String,
+            service_account_id: String,
+            id: String,
+        }
+
+        let key: JsonKey = serde_json::from_str(json_key)?;
+
+        Ok(Self {
+            key_id: key.id,
+            service_account_id: key.service_account_id,
+            private_key: SecretString::new(key.private_key),
+        })
+    }
+
+    const AUDIENCE: &'static str = "https://iam.api.cloud.yandex.net/iam/v1/tokens";
```

Comment:
Please make it as config param for `ServiceAccountCredentials` with the default value.
IAM service may have more, then one installation/endpoint, similar to [google/yandex metadata request uri](https://github.com/ydb-platform/ydb-rs-sdk/blob/main/ydb/src/credentials.rs#L333-L333).

Reply:
I maked this value as parameter with default value. But little bit different from metadata, because there is multiple methods for creating instance. I added separate method to specify custom url:

```
 pub fn with_url(mut self, url: impl Into<String>) -> Self {
     self.audience_url = url.into();
     self
}
```